sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -454,6 +515,12 @@ private void performLogging(LogLevel logLevel, boolean isExceptionLogging, Strin
      * @param args Arguments for the message, if an exception is being logged last argument is the throwable.
      */
     private void performDeferredLogging(LogLevel logLevel, Supplier<String> messageSupplier, Throwable throwable) {
+        if (!globalContextSerialized.isEmpty()) {
+            LoggingEventBuilder.create(logger, logLevel, globalContextSerialized, canLogAtLevel(logLevel))
+                .log(messageSupplier.get(), throwable);
```

Comment:
Is this eagerly getting the message, even in cases when `canLogAtLevel(logLevel)` returns false? Should you be calling `.log(messageSupplier, throwable)` instead?

Reply:
this method is only called when logging is enabled, but you're right, it's still better to call `log` overload with the supplier.